controllers/cluster_controller_test.go

```diff
@@ -381,11 +372,6 @@ func newAzureClusterWithControlPlane() (*capi.Cluster, *kcp.KubeadmControlPlane,
 
 func newAzureMachinePool(cluster, name string) *capzexp.AzureMachinePool {
 	return &capzexp.AzureMachinePool{
-		// TODO (mig4): I think specifying TypeMeta explicitly is only necessary because
-		//   we don't start the manager correctly, this causes references to this object
-		//   be invalid because when `reference.GetReference` sees an empty GVK it tries
-		//   to look it up but fails because testenv is not fully initialised; i.e. test
-		//   if this is necessary if we initialise the manager in `suite_test.go`
 		TypeMeta: metav1.TypeMeta{
```

Comment:
Do we still need the `TyeMeta` here?

Reply:
For now yes, as I split the envtest stuff into a separate PR - #17 as it's not quite ready yet.